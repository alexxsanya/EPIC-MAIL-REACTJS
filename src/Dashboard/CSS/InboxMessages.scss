@import '../../common/CSS/assets/variables';
@import '../../common/CSS/assets/fonts';

.inbox-messages {

  table {
    font-family: arial, sans-serif;
    border-collapse: collapse;
    width: 100%;
    font-size: 0.90em;
    min-width: 900px;
  }

  tr {
    background: $white;
    border-bottom: 1px solid #eae2e2;
    color: #555;
  }

  tr:hover{
    background: $primary-blue;
    color: $white;
    cursor: pointer;
  }

  td, th {
    text-align: left;
    padding: 8px;
  }

  td.td-action{
    color:#bbbbbb;
  }

  td.td-action:hover{
    color:#f91212;
  }

  td.td-action.positive:hover{
    color:#fff;
  }

}