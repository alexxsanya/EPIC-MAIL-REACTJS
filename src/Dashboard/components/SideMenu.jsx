import React from 'react';
import {
  Nav,
} from 'react-bootstrap';

const SideMenu = () => (
  <Nav defaultActiveKey="/home" className="flex-column">
    <Nav.Link href="/dashboard/inbox">Inbox Messages</Nav.Link>
    <Nav.Link href="/dashboard/sent">Sent Messages</Nav.Link>
  </Nav>
);
export default SideMenu;
