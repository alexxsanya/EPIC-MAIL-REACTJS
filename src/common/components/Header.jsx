import React from 'react';
import {
  Navbar,
  Image,
  ButtonToolbar,
  DropdownButton,
  Dropdown,
} from 'react-bootstrap';
import { Link } from 'react-router-dom';
import user from '../assets/images/user.png';
import '../CSS/Header.scss';

const token = localStorage.getItem('user_token');
const username = localStorage.getItem('username');
export default function NavSection() {
  return (
    <header>
      <Navbar className="nav-section">
        <Navbar.Brand href="/">
          EPIC MAIL
        </Navbar.Brand>
        <Navbar.Toggle />
        <Navbar.Collapse className="justify-content-end nav-auth-section">
          {
            token
              ? (
                <>
                  <Navbar.Text variant="light" className="profile-photo">
                    <Image src={user} roundedCircle width="40" height="40" />
                  </Navbar.Text>
                  <ButtonToolbar>
                    <DropdownButton
                      drop="down"
                      title=""
                      id="dropdown-button-drop-down"
                      key="down"
                    >
                      <Dropdown.Item className="profile-name" eventKey="1">{username}</Dropdown.Item>
                      <Dropdown.Item eventKey="2" href="/profile">Profile</Dropdown.Item>
                      <Dropdown.Item eventKey="2" href="/profile">Dasboard</Dropdown.Item>
                      <Dropdown.Divider />
                      <Dropdown.Item
                        eventKey="4"
                        href="#"
                        onClick={() => {
                          localStorage.removeItem('user_token');
                          window.location = '/';
                        }}
                      >
                          Logout
                      </Dropdown.Item>
                    </DropdownButton>
                  </ButtonToolbar>
                </>
              )
              : (
                <>
                  <Link to="/login">
                    <Navbar.Text variant="light">
                  Login &nbsp;|
                    </Navbar.Text>
                  </Link>
                  <Link to="/signup">
                    <Navbar.Text variant="light">
                  &nbsp; Signup
                    </Navbar.Text>
                  </Link>
                </>
              )
          }
        </Navbar.Collapse>
      </Navbar>
    </header>
  );
}
