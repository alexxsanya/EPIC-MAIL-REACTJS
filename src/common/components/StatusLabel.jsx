/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import '../CSS/StatusLabel.scss';
import { Alert } from 'react-bootstrap';
import PropTypes from 'prop-types';

function StatusLabel({ message, tip, variant }) {
  return (
    <Alert variant={variant} className="status-box">
      {message}
      ,
      &nbsp;
      {tip}
    </Alert>
  );
}

StatusLabel.propTypes = {
  message: PropTypes.string,
  tip: PropTypes.string,
  variant: PropTypes.string,
};


StatusLabel.defaultProps = {
  message: 'Hi ',
  tip: 'Try refreshing the page.',
  variant: 'warning',
};

export default StatusLabel;
