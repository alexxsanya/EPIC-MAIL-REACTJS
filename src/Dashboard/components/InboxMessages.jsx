/* eslint-disable no-shadow */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Table } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import StatusLabel from '../../common/components/StatusLabel';
import '../CSS/InboxMessages.scss';
import fetchInboxMessages from '../Redux/actions/InboxMessagesAction';

export class InboxMessages extends React.Component {
  componentWillMount() {
    const { fetchInboxMessages } = this.props;
    fetchInboxMessages();
  }

  render() {
    const { messages, inboxFetched } = this.props;

    return (
      <div className="inbox-messages">
        <h2 className="section-title">Inbox Messages</h2>
        {
        inboxFetched
          ? (
            <div className="messages-display-box">
              <Table responsive>
                <tbody>
                  {
                  messages.reverse().map(message => (
                    <tr key={message.id} className={message.readstatus}>
                      <td>{message.subject}</td>
                      <td>{message.msgbody}</td>
                      <td>{message.createdon}</td>
                      <td> </td>
                    </tr>
                  ))
                }
                </tbody>
              </Table>
            </div>
          )
          : (
            <StatusLabel
              message="Oops"
              tip="You currently have no Inbox Message"
            />
          )
      }
      </div>
    );
  }
}

InboxMessages.propTypes = {
  messages: PropTypes.array.isRequired,
  inboxFetched: PropTypes.bool.isRequired,
  fetchInboxMessages: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  isFetching: state.inboxReducer.isFetching,
  errorOccured: state.inboxReducer.error,
  inboxFetched: state.inboxReducer.status,
  messages: state.inboxReducer.items,
  response: state.inboxReducer.message,
});

export default connect(mapStateToProps, { fetchInboxMessages })(InboxMessages);
