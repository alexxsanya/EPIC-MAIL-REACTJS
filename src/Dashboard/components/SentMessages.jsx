/* eslint-disable no-shadow */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Table } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import StatusLabel from '../../common/components/StatusLabel';
import '../CSS/InboxMessages.scss';
import fetchtOutboxMessages from '../Redux/actions/SentMessagesAction';

class SentMessages extends React.Component {
  componentWillMount() {
    const { fetchtOutboxMessages } = this.props;
    fetchtOutboxMessages();
  }

  render() {
    const { messages, sentFetched } = this.props;
    console.log(messages);
    return (
      <div className="outbox-messages">
        <h2 className="section-title">Sent Messages</h2>
        {
        sentFetched
          ? (
            <div className="messages-display-box">
              <Table responsive>
                <tbody>
                  {
                  messages.map(message => (
                    <tr key={message.id} className={message.readstatus}>
                      <td>{message.subject}</td>
                      <td>{message.msgbody}</td>
                      <td>{message.createdon}</td>
                      <td> </td>
                    </tr>
                  ))
                }
                </tbody>
              </Table>
            </div>
          )
          : (
            <StatusLabel
              message="Oops"
              tip="You currently haven't sent anyone a message"
            />
          )
      }
      </div>
    );
  }
}

SentMessages.propTypes = {
  messages: PropTypes.array.isRequired,
  sentFetched: PropTypes.bool.isRequired,
  fetchtOutboxMessages: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  isLoggingIn: state.outboxReducer.isFetching,
  errorOccured: state.outboxReducer.error,
  sentFetched: state.outboxReducer.status,
  messages: state.outboxReducer.items,
  response: state.outboxReducer.message,
});

export default connect(mapStateToProps, { fetchtOutboxMessages })(SentMessages);
