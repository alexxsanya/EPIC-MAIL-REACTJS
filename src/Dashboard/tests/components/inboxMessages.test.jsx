/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { InboxMessages } from '../../components/InboxMessages';

const props = {
  isFetching: false,
  errorOccured: {
    errorMessage: 'No Message',
  },
  inboxFetched: false,
  response: 'No Message',
  messages: [],
  fetchInboxMessages: jest.fn(),
};

describe('InboxMessages Component', () => {
  const component = shallow(<InboxMessages {...props} />);

  it('should match the snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  // it('should call fetchInboxMessage on Component Will mount', () => {
  //   const wrapper = mount(<InboxMessages />);
  //   wrapper.setProps(props);
  //   expect(wrapper).toBeDefined();
  //   expect(props.fetchInboxMessages()).toHaveBeenCalled();
  // });
});
