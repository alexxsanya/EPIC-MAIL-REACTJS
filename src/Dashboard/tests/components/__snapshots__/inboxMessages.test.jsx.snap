// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InboxMessages Component should match the snapshot 1`] = `
<div
  className="inbox-messages"
>
  <h2
    className="section-title"
  >
    Inbox Messages
  </h2>
  <StatusLabel
    message="Oops"
    tip="You currently have no Inbox Message"
    variant="warning"
  />
</div>
`;
